docs/features/module-initializers.md

```diff
@@ -0,0 +1,109 @@
+# Module Initializers
+## Overview
+
+A module initializer is a method that runs eagerly and one-time when an assembly is loaded, similar to a type initializer (static constructor).
+
+A module initializer in .NET is defined by adding a `.cctor` method to the `<Module>` type. Since the `<Module>` type is not explicitly defined in user code, the language has to provide a feature in order to make it possible for users to write a module initializer.
+
+__See__: the [corresponding proposal](https://github.com/dotnet/csharplang/blob/master/proposals/module-initializers.md) in CSharpLang.
+
+## Defining a module initializer method
+
+A method can be designated as module initializer by decorating it with a `[ModuleInitializer]` attribute.
+
+```cs
+using System;
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    public class ModuleInitializerAttribute : Attribute { }
+}
+```
+
+The attribute can be used like this:
+
+```cs
+using System.Runtime.CompilerServices;
+
+class C
+{
+    [ModuleInitializer]
+    internal static void M1()
+    {
+        // ...
+    }
+}
+```
+
+Some requirements are imposed on the method targeted with this attribute:
+1. The method must be `static`.
+1. The method must be parameterless.
+1. The method must return `void`.
+1. The method must be accessible from the `<Module>` class.
+    - This means the method's effective accessibility must be `internal` or `public`.
+    - This also means the method cannot be a local function.
```

Comment:
(Comparing against https://github.com/dotnet/roslyn/pull/43301/files#diff-eac3afc2449ebcd568f92bea7d52d604)

```markdown
1. The method must not be generic or declared in a generic type.
```

Is it worth pointing out that module initializers must be ordinary methods, as opposed to finalizers, accessors, and operators (and local functions) which are allowed by AttributeTargets.Method?